### ISSUE TITLE
Search results webpart - Custom results layout: pnp-details-list inside group

### ISSUE THREAD

**Issue by ileana2019** [microsoft-search/pnp-modern-search#460]
**Version used**
3.14.1.0

**Describe the bug**
I created a custom .html results layout containing a web component pnp-details-list inside a group:

```
{{#group items by="Site"}}
<pnp-details-list data-items="{{JSONstringify items}}" data-columns-configuration="[{&quot;uniqueId&quot;:&quot;986cc099-2e5f-47e9-8b3e-aaa4bcaf8f50&quot;,&quot;name&quot;:&quot;Actions&quot;,&quot;value&quot;:&quot;{%replace ParentLink 'AllItems' 'DispForm'%}&quot;,&quot;useHandlebarsExpr&quot;:true,&quot;minWidth&quot;:&quot;70&quot;,&quot;maxWidth&quot;:310,&quot;enableSorting&quot;:null,&quot;isResizable&quot;:null,&quot;isMultiline&quot;:null,&quot;isResultItemLink&quot;:null,&quot;sortIdx&quot;:1}]" data-show-file-icon="true" data-is-compact=""></pnp-details-list>
{{/group}}
```

data-columns-configuration corresponds to this JSON:
```
[{
		"uniqueId": "986cc099-2e5f-47e9-8b3e-aaa4bcaf8f50",
		"name": "Actions",
		"value": "{{replace ParentLink 'AllItems' 'DispForm'}}",
		"useHandlebarsExpr": true,
		"minWidth": "70",
		"maxWidth": 310,
		"enableSorting": null,
		"isResizable": null,
		"isMultiline": null,
		"isResultItemLink": null,
		"sortIdx": 1
	}]
```

The document title is not displayed inside the pnp-details-list, because (I assume) of the double curly braces in the value: "value": "{{replace ParentLink 'AllItems' 'DispForm'}}". 

Is there a way to convert the "{{" into a string in the DOM?

**To Reproduce**
1. In the search results webpart properties, page 3, select the Results layout: Custom
2. Use an external template URL: pointing to a .html file location in SharePoint
3. Edit the .html file for the results layout to include a group and a pnp-details-list (see example in the description)
4. When using a handlebar expression for 1 field, the value property is empty
![Result](https://user-images.githubusercontent.com/55545989/92405943-0b1a7500-f137-11ea-8244-125ec01c746f.PNG)

**Expected behavior**
Display the handlebar expression inside the value, in the DOM.
![Expected result](https://user-images.githubusercontent.com/55545989/92405837-cc84ba80-f136-11ea-839c-8ceb5232c822.PNG)

**Desktop (please complete the following information):**
 - Browser [e.g. chrome]

**Comment by wobba** [comment]
Question, what kind of link are you trying to create by replacing AllItems.aspx with DispForm.aspx? And I'm no handlebars expert, so might help to ask in a general handlebar forum as well.

**Comment by ileana2019** [comment]
Hello,
I would like to create a View/Edit document properties button inside the pnp-details-list. 

When selecting "Details list" in the "Results layout", and then clicking on "Edit template", this is how the pnp-details-list is configured:
`<pnp-details-list data-items="{{JSONstringify items}}" data-columns-configuration="{{JSONstringify detailsListColumns}}" data-show-file-icon="{{@root.showFileIcon}}" data-is-compact="{{@root.isCompact}}"></pnp-details-list>`
When selecting "Custom" in the "Results layout" and then edit the results template to add a pnp-details-list with data-columns-configuration = the above JSON with handlebar expression, it doesn't work. And I think it is because of the double curly braces which are interpreted as a code and not as string inside the template.

I will contact also the handlebar forum.

**Comment by wobba** [comment]
Can you share the full custom template?

**Comment by ileana2019** [comment]
Yes, please find the template attached.
Thanks!
PS: I also posted on the handlebar helpers issues list (no reply yet)
[CustomTemplate.txt](https://github.com/microsoft-search/pnp-modern-search/files/5187882/CustomTemplate.txt)

**Comment by wobba** [comment]
You probably want to delete the extra markup at the bottom of your template ;) 

I managed to make it work I believe by doing this.

1. Pick "Details list as the layout"
2. Edit the column set up like you want. I added this to one:
`{{replace ParentLink 'AllItems' 'DispForm'}}`
![image](https://user-images.githubusercontent.com/51104/92505959-045a3380-f205-11ea-948a-cfb1ffb9babf.png)
4. Link to a template or paste in the web part.
3. Used this in my group by
```
{{#group items by="contentclass"}}
   <h1>{{value}}</h1>
   <pnp-details-list data-items="{{JSONstringify items}}" data-columns-configuration="{{JSONstringify @root.detailsListColumns}}" data-show-file-icon="true" data-is-compact=""></pnp-details-list>
{{/group}}
```

This way the column setup is saved as part of the web part, and used by the custom template. Makes sense? The array in your sample template is not a full column config including all columns either. If you want to do it your way you must instrument the full `detailsListColumns` list.
